### PR TITLE
Minor fixes for output writer

### PR DESCRIPTION
**Pull request by joshgav**
Mostly minor adjustments for output. The main commit is the first one - create the full directory path (mkdir -p) if it doesn't yet exist.

### CODE REVIEW

**Comment by piotrpMSFT** [comment]
@joshgav please add unit tests to Given_a_Bootstrapper for the behaviors you are introducing.

**Comment by joshgav** [comment]
Will do. Not surprisingly, I found this error in Jenkins:
```
Generating Client Library to d:\j\workspace\microsoft_vipr_prtest\test\ViprCliUnitTests\bin\Debug\output
Done.
z:\Builds\work\b7ceef20fdfcf0ee\Shared\Execution\FallbackTestFramework.cs(21): error : ViprCliUnitTests.Given_a_Bootstrapper.When_custom_outputPath_is_not_specified_then_defaults_are_used: FluentAssertions.Execution.AssertionFailedException : Expected True Because the proxy was created in the working directory., but found False. [d:\j\workspace\microsoft_vipr_prtest\test.proj]
```

Should probably have tests for when output path:
- isn't specified (default)
- specified as empty string `""`
- specified as single folder (e.g. ".\mycode")
- specified as multi-level folder (e.g. ".\mycode\csharp")

**Comment by MrTomWhite** [comment]
Commits 126791b ~ f4c4b13 could probably be squashed together. (They all address the same file)

**Comment by joshgav** [comment]
I believe there are assumptions in Given_a_bootstrapper.ValidateProxyConfiguration that output will be in the current working directory, as FileSystemHelpers.ValidateTextFiles is configured by default to look in the working directory. In the command lines passed in to ValidateProxyConfiguration currently that is the case, but it may not always be. The output directory should be parsed from the command line passed to ValidateProxyConfiguration, or passed as a separate parameter, and then specified directly in FileSystemHelpers.ValidateTextFiles.

File: src/Core/Vipr/Bootstrapper.cs [hunk 1]
```diff
@@ -30,15 +30,17 @@ internal class Bootstrapper
         private string _readerName = "ODataReader.v4";
         private string _writerName = "CSharpWriter";
         private string _metadataPath = "http://services.odata.org/V4/TripPinServiceRW/$metadata";
-        private string _outputPath = "";
```
Comment (piotrpMSFT) [inline src/Core/Vipr/Bootstrapper.cs]: How do we specify that we want to use the current directory as the output path? Can we have a test for this?

File: test/ViprCliUnitTests/Given_a_Bootstrapper.cs [hunk 2]
```diff
@@ -158,14 +161,16 @@ public void When_custom_outputPath_is_not_specified_then_defaults_are_used()
...
+                File.Exists(pathToOneExpectedOutputFile).Should().BeTrue("Because one expected output file was created in default output directory.");
 
-                if (File.Exists("CSharpProxy.cs")) File.Delete("CSharpProxy.cs");
+                if (Directory.Exists(defaultOutputDirectory)) Directory.Delete(defaultOutputDirectory, true);
             });
         }
 
         [Fact]
-        public void When_custom_outputPath_is_specified_then_defaults_are_used()
+        public void When_custom_outputPath_is_specified_then_it_is_used()
```
Comment (piotrpMSFT) [inline test/ViprCliUnitTests/Given_a_Bootstrapper.cs]: Great catch here!

**Comment by piotrpMSFT** [comment]
@joshgav you mentioned we should have tests for:
Should probably have tests for when output path:

* isn't specified (default)
* specified as empty string ""
* specified as single folder (e.g. ".\mycode")
* specified as multi-level folder (e.g. ".\mycode\csharp")

but I think the PR does not include tests for

* specified as empty string ""
* specified as multi-level folder (e.g. ".\mycode\csharp")

Can you please add?

File: src/Core/Vipr/Bootstrapper.cs (hunk 1 above)
Comment (joshgav) [inline src/Core/Vipr/Bootstrapper.cs]: empty string or dot - e.g. `--outputPath=""` or `--outputPath="."` or `--outputPath=".\"`
I added a test for the empty string case.

**Comment by joshgav** [comment]
Done!

File: test/ViprCliUnitTests/ViprCliUnitTests.playlist [hunk 3]
```diff
@@ -0,0 +1 @@
...iprCliUnitTests.Given_a_Bootstrapper.When_custom_outputPath_is_specified_then_it_is_used" /><Add Test="ViprCliUnitTests.Given_a_Bootstrapper.When_a_custom_writer_is_specified_then_it_is_used" /><Add Test="ViprCliUnitTests.Given_a_Bootstrapper.When_a_custom_reader_is_specified_then_it_is_used" /><Add Test="ViprCliUnitTests.Given_a_Bootstrapper.When_modelexport_is_set_it_exports_the_model" /><Add Test="ViprCliUnitTests.Given_a_Bootstrapper.When_a_reader_implements_IConfigurable_then_SetConfigurationProvider_is_called" /><Add Test="ViprCliUnitTests.Given_a_Bootstrapper.When_two_level_outputPath_is_specified_then_it_is_used" /><Add Test="ViprCliUnitTests.Given_a_Bootstrapper.When_a_writer_implements_IConfigurable_then_SetConfigurationProvider_is_called" /><Add Test="ViprCliUnitTests.Given_a_FileWriter.When_outputDirectoryPath_is_not_specified_then_it_writes_the_files_to_the_working_directory" /><Add Test="ViprCliUnitTests.Given_a_Bootstrapper.When_empty_string_outputPath_is_specified_then_current_working_dir_is_used" /><Add Test="ViprCliUnitTests.Given_a_FileWriter.When_outputDirectoryPath_is_specified_and_does_not_exist_then_it_creates_the_directory_and_writes_the_files" /></Playlist>
```
Comment (piotrpMSFT) [inline test/ViprCliUnitTests/ViprCliUnitTests.playlist]: We don't use test playlists in Vipr. Any reason we need to add one here?

**Pull request merged**